internal/state/state.go

```diff
@@ -196,3 +197,28 @@ var StopListeners func() error
 
 // ReloadClusterCert reloads the cluster keypair from the state directory.
 var ReloadClusterCert func() error
+
+// Unrestrict returns the underlying InternalState from the exposed State interface.
+func Unrestrict(s State) (*InternalState, error) {
```

Comment:
As there really isn't any restriction applied, I guess it would be better to rename this to something like `Resolve` or `ResolveInternalState`? Or `Derive`/`DeriveInternalState`.

Reply:
I've called it `ToInternal` now.